src/components/Button.tsx

```diff
@@ -1,3 +1,72 @@
-export const FirstButton = () => {
-    return <div />
-}
\ No newline at end of file
+import { BiGame, BiRocket, BiSmile } from "react-icons/bi";
+
+type Icon = 'rocket' | 'smiley' | 'game'
+export interface ButtonProps {
+    variant?: 'primary' | 'secondary' | 'tertiary';
+    size?: 'small' | 'medium' | 'large';
+    disabled?: boolean;
+    loading?: boolean;
+    onClick: () => void;
+    text: string;
+    icon?: Icon;
+    iconPosition?: 'left' | 'right';
+    // tooltip?: string;
+}
+
+export const Icon = ({ name }: { name: Icon }) => {
+    if (name === 'rocket') {
+        return <BiRocket />
+    } else if (name === 'smiley') {
+        return <BiSmile />
+    } else if (name === 'game') {
+        return <BiGame />
+    }
+}
+
+export const FirstButton = ({ variant = 'primary', size = 'medium', disabled = false, loading = false, onClick, text, icon, iconPosition = 'left' }: ButtonProps) => {
+
+    const baseClasses = 'flex justify-center items-center px-4 py-2 font-bold text-white bg-blue-500 rounded-md hover:bg-blue-700 focus:outline-none transition-scale duration-150 ease-in-out transform hover:scale-[102%] active:scale-[97%]'
+
+    const variantClasses = {
+        primary: 'text-white bg-blue-500 hover:bg-blue-700',
+        secondary: 'text-gray-500 bg-gray-500 hover:bg-gray-800',
+        tertiary: 'text-gray-800 bg-transparent hover:bg-gray-100',
```

Comment:
don't use bg-transparent for a button - it would be very unpredictable when placed above non-white elements, might be seen as malicious by crawlers etc